crates/nu-path/src/pwd_per_drive.rs

```diff
@@ -0,0 +1,349 @@
+    }
+}
+
+struct DriveToPwdMap {
+    map: [Option<String>; 26], // Fixed-size array for A-Z
+}
+
+impl DriveToPwdMap {
+    pub fn new() -> Self {
+        DriveToPwdMap {
+            map: Default::default(), // Initialize all to `None`
+        }
+    }
+
+    /// Set the PWD for the drive letter in the absolute path.
+    /// Return String for error description.
+    pub fn set_pwd(&mut self, path: &Path) -> Result<(), PathError> {
+        if let (Some(drive_letter), Some(path_str)) =
+            (Self::extract_drive_letter(path), path.to_str())
+        {
+            if drive_letter.is_ascii_alphabetic() {
+                let drive_letter = drive_letter.to_ascii_uppercase();
+                // Make sure saved drive letter is upper case
+                let mut c = path_str.chars();
+                match c.next() {
+                    None => Err(PathError::InvalidDriveLetter),
+                    Some(_) => {
+                        let drive_index = drive_letter as usize - 'A' as usize;
+                        self.map[drive_index] = Some(drive_letter.to_string() + c.as_str());
+                        let _ = std::env::set_current_dir(path_str);
+                        Ok(())
+                    }
+                }
+            } else {
+                Err(PathError::InvalidDriveLetter)
+            }
+        } else {
+            Err(PathError::InvalidPath)
+        }
+    }
+
+    /// Get the PWD for drive, if not yet, ask GetFullPathNameW(),
+    /// or else return default r"X:\".
+    fn get_pwd(&mut self, drive_letter: char) -> Result<String, PathError> {
+        if drive_letter.is_ascii_alphabetic() {
+            let drive_letter = drive_letter.to_ascii_uppercase();
+            let drive_index = drive_letter as usize - 'A' as usize;
+            Ok(self.map[drive_index].clone().unwrap_or_else(|| {
+                if let Some(sys_pwd) = get_full_path_name_w(&format!("{}:", drive_letter)) {
+                    self.map[drive_index] = Some(sys_pwd.clone());
+                    sys_pwd
+                } else {
+                    format!(r"{}:\", drive_letter)
+                }
+            }))
+        } else {
+            Err(PathError::InvalidDriveLetter)
+        }
+    }
+
+    /// Expand a relative path using the PWD-per-drive, return PathBuf
+    /// of absolute path.
+    /// Return None if path is not valid or can't get drive letter.
+    pub fn expand_path(&mut self, path: &Path) -> Option<PathBuf> {
+        if need_expand_pwd_per_drive(path) {
+            let path_str = path.to_str()?;
+            if let Some(drive_letter) = Self::extract_drive_letter(path) {
+                if let Ok(pwd) = self.get_pwd(drive_letter) {
+                    // Combine current PWD with the relative path
+                    let mut base = PathBuf::from(Self::ensure_trailing_delimiter(&pwd));
+                    base.push(path_str.split_at(2).1); // Skip the "C:" part of the relative path
```

Comment:
i worry about this split_at(2). 
1. what happens with unc paths?
2. is there a omnipath function to handle this better?

Reply:
Eliminate 'split_at(2)', and add comment to declare safety to extracting parts after index 2.